185799263 add regression ci

tejal-shah opened the pull request.

tealefristoe commented on cypress/e2e/clue/branch/student_tests/dataflow_tool_spec.js:
```diff
@@ -11,10 +11,11 @@ context('Dataflow Tool Tile', function () {
   });
   describe("Dataflow Tool", () => {
     it("renders dataflow tool tile", () => {
+      cy.wait(120000);
```
2 minute wait on every run 😱 

I know this test has caused a lot of flakiness, so maybe it's best to be super conservative and make sure it has a chance to load. But why does it take 2 minutes to load? To me, this feels like a temporary work around that doesn't address the real underlying problem, though I'll freely admit that I don't know what the real problem is or how to fix it.

tejal-shah replied:
For some reason, the default problem document in the `dfe` unit takes really long to load in github (This doesn't happen in real life and it's also not all that bad in a local cypress test) - and almost always it is the initial load time of the default problem document that has caused the test to fail. I tried a 1 min timeout and that didn't improve things. So, with this PR, I'm trying two things: 
- moving the check to see if the document loads into the test itself from the `before` hook so that it gets retries when it fails (Cypress doesn't retry code that fails in the `before` and `after` hooks)
- increasing the load time to 2 min. It's not ideal, but it is only for this one test and not everywhere and since this test runs in parallel with other ones, my hope is that this wait will not much affect the overall test running times. I've also increased the number of threads that the test job uses from 5 to 7 which has improved the average test run times from  30 min to 20 min.

tealefristoe commented on .github/workflows/regression.yml:
```diff
@@ -13,11 +13,54 @@ on:
+      test:
+        description: Tests to run
+        default: all
+        options:
```
Is this something we should add to when creating new tests?

tejal-shah replied:
If we want the ability to run that specific newly added spec file, then yes.

scytacki commented on .github/workflows/ci.yml:
```diff
@@ -112,4 +128,60 @@ jobs:
+      # https://github.com/cypress-io/github-action/issues/48
+      fail-fast: false
+      matrix:
+        # run 5 copies of the current job in parallel
```
this comment is out of date with the 7 containers

scytacki reviewed (approved):
This looks good to me. I just found one out of date comment. You don't need to re-request a review.

After this is merged we need to update the branch protection rules since they are currently making sure all 5 of the cypress jobs have passed before allowing to merge. Now they only need to check for the 1 cypress job.

tealefristoe reviewed (commented):
There's a lot in here that I don't understand, but I appreciate the opportunity to get a glimpse at how this is all set up. It looks ok to me but I'll let someone with more knowledge give it the official 👍 

From what I understood, my biggest concern is seeing a couple of two minute waits being added. Seems like one (line 221 of `commands.js`) won't always take two minutes, but the `cy.wait` added to `dataflow_tool_spec.js` will always take two minutes. This seems like a direction we want to avoid going, but maybe it's unavoidable.

tejal-shah commented:
@scytacki , I changed the required checks from cypress(5) to cypress(1) when I changed the builds to use only the smoke tests that use a single thread.
